### PR TITLE
Removing environment specific urls

### PR DESCRIPTION
**Pull request by JohnPTobe**
Replacing environment specific urls with environment variables

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,15 +1,15 @@
-FROM docker.platform.cloud.coe.ic.gov/centos7
+FROM $REGISTRY/centos7
```
Comment (gisjedi) [inline Dockerfile]: An ARG statement will be needed prior to the FROM statement to allow this to build. I'd parameterize the whole thing, not just the registry. Ref: https://github.com/ngageoint/scale/blob/a99f2e5f3792e2f6cc9c1e129596c7e48c3beedd/Dockerfile#L1

File: Dockerfile [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
+    RUN="docker run -d -p 9000:9000 -p 80:80 -v <silo db/log location>:/usr/silo silo" \
+    SOURCE="https://github.com/ngageoint/seed-silo" \
     DESCRIPTION="seed-silo api" \
     CLASSIFICATION="UNCLASSIFIED"
 
 RUN mkdir -p silo \
     && yum install wget -y \
     && update-ca-trust enable \
-    && wget http://pki-ldap.ismc.ic.gov/ -r -A *.cer -nd -nv -P /etc/pki/ca-trust/source/anchors/ \
+    && wget $PKIURL -r -A *.cer -nd -nv -P /etc/pki/ca-trust/source/anchors/ \
```
Comment (gisjedi) [inline Dockerfile]: This should all be offloaded into the base image so I'd just remove this entire RUN statement.

File: build-docker.sh [hunk 3]
```diff
@@ -21,7 +21,7 @@ esac
 build-silo.sh
 
 ${SUDO} docker build . -t silo:$VERSION
```
Comment (gisjedi) [inline build-docker.sh]: You'll need to call `docker build` with build arg to specify the base image here. This is must be passed on from whatever environment is provided from the build system (Gitlab / Travis / etc.).

File: build-docker.sh [hunk 4]
```diff
@@ -21,7 +21,7 @@ esac
...
-${SUDO} docker tag silo:$VERSION docker.platform.cloud.coe.ic.gov/nga-r-dev/silo:$VERSION
```
Comment (gisjedi) [inline build-docker.sh]: Not really any need to tag individually here... might as well just do that in the build step.

File: build-docker.sh [hunk 5]
```diff
@@ -12,6 +12,20 @@ then
     VERSION=snapshot
 fi
 
+REGISTRY=$1
```
[the remark this answers is not in the document]
Comment (JohnPTobe) [reply build-docker.sh]: That's for using environment variables, isn't it? I'm using passed in args to the shell script.  I don't have a problem with switching to environment variables if that's better.

[the remark this answers is not in the document]
Comment (gisjedi) [reply build-docker.sh]: Well, $1 is an environment variable as well, albeit a built-in one for capturing positional command-line args. You could use the following mechanism to place in `REGISTRY` in one shot:

```
REGISTRY=${1:-docker.io}
```

That will use $1 if non-empty and if not use `docker.io`. Note the significant difference between `-` and `=` in the parameter expansion syntax. `-` provides expansion with a default, `=` actually also assigns back into the source environment variable if unset.

**Comment by gisjedi** [comment]
We should go ahead and get a gitlab build here. You can use the Scale one as a reference: https://github.com/ngageoint/scale/blob/master/.gitlab-ci.yml

You can set up a mirror to Gitlab to test the builds. I have that for Scale to test Gitlab builds and just created one for silo. You should have an invite for the group.

**Comment by JohnPTobe** [comment]
@gisjedi It took awhile, but the gitlab pipeline is finally working.  Unless there are other changes that need to be made, I think we're ready to merge this branch.

**Pull request merged**